### DISCUSSION TITLE
Can the package @wordpress/api-fetch also be used on the frontend?

### DISCUSSION THREAD

**Discussion by ivanjeremic** [WordPress/gutenberg#38137] [Q&A] [answered]
Can we use `@wordpress/api-fetch` also for front-end code when we are building client side rendered blocks? Also is using alternatives like Axios fine?

**Comment by gziolo** [comment, marked as answer]
Yes, you should be able to use `@wordpress/api-fetch` also on the front end. It requires some configuration but you can check how it's done in the admin here:

https://github.com/WordPress/wordpress-develop/blob/b8f97013df4fb461c72ebf7d0b5166bac15a526b/src/wp-includes/script-loader.php#L294-L323

I don't know how much of that is related to the block editor so you need to find your ideal setup. 

>  Also is using alternatives like Axios fine?

I guess it's also fine. In particular, when you only want to perform requests for logged out users where handling for nonces would be useful.
